rfcs/0044-node-status-backend.md

```diff
@@ -0,0 +1,55 @@
+## Summary
+
+[summary]: #summary
+
+This RFC proposes that we use AWS stack to be the backend of the node status/error system. To be specific by AWS stack, I mean we would use S3 as the storage backend; OpenSearch as the search engine; Kibana as the visualization and charting tool; we could also setup the miniservice that listening on the minaprotocol/node-status on AWS ES2.
+
+## Motivation
+
+[motivation]:#motivation
+
+We need a backend for node staus/error systems. Candidates for the backend include AWS stack and Grafana Loki and LogDNA.
+
+## Implementation
+
+[implementation]:#implementation
+
+A mini-service will be deployed at https://node-status.minaprotocol.com. The mini-service would do some simple check against the format of the data that nodes send us. The mini-service would be implemented using python or javascript. The mini-service would listen on the designated port and decode the json data that peers send and then do a simple check that make sure that all the non-optional field are present and then call AWS `PUT` function to push it to the AWS kenesis firehose data stream. This mini-service would be put in a AWS EC2 container to make the configuration of things minimal.
+
+I only consider simple validity checks here against the format of the json data because any more complicated check would require us to run a node to observe the network conditions.
+
+For the AWS stack we need to setup
+1 Kenesis firehose data stream to receive the logs, and
+1 S3 storage bucket to store the logs, and
+1 OpenSearch service that provides the search ability for the logs, and
+1 Kibana service that provides the visualization for the data
```

Comment:
Could we make this RFC a bit broader to accomodate the backend requirements for the node error collection backend as well? They are very similar and I want us to have one common flexible system for both contexts.  In such scenario, I would assume we would basically duplicate the described setup for the error collection backend system, is my understanding correct?

Reply:
Yes. We would just duplicate the setup for node error system too.